summingbird-online/src/main/scala/com/twitter/summingbird/online/executor/AsyncBase.scala

```diff
@@ -107,13 +136,15 @@ abstract class AsyncBase[I, O, S, D, RC](maxWaitingFutures: MaxWaitingFutures, m
     if (pendingFuturesCount > maxWaitingFuturesCount) {
       // Too many futures waiting, let's clear.
       outstandingFutures.dequeueAll(_.isDefined)
```

Comment:
I also noticed another issue which earlier implementation was accidentally avoiding. Since we'll now unblock immediately when we are below the limit we'll likely hit the limit pretty soon and then the high cost of iterating over all futures will come into picture. I think we should treat the limit as a high watermark and bring down the number of pending futures by x%, say 10%. This way we can amortize the cost of these operations. What do you think?